stdlib/public/core/Span/RawSpan.swift

```diff
@@ -0,0 +1,651 @@
+  /// `pointer` must be owned by the instance `owner`,
+  /// meaning that as long as `owner` is alive the memory will remain valid.
+  ///
+  /// - Parameters:
+  ///   - pointer: a pointer to the first initialized byte.
+  ///   - byteCount: the number of initialized bytes in the span.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `RawSpan`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  public init<T: BitwiseCopyable>(
+    _unsafeStart pointer: UnsafePointer<T>,
+    count: Int
+  ) {
+    _precondition(count >= 0, "Count must not be negative")
+    self.init(
+      _unchecked: pointer, byteCount: count*MemoryLayout<T>.stride
+    )
+  }
+
+  /// Create a `RawSpan` over the memory represented by a `Span<T>`
+  ///
+  /// - Parameters:
+  ///   - span: An existing `Span<T>`, which will define both this
+  ///           `RawSpan`'s lifetime and the memory it represents.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @unsafe // remove when fixing the lifetime annotation
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  public init<Element: BitwiseCopyable>(
+    _unsafeSpan span: borrowing Span<Element>
+  ) {
+    self.init(
+      _unchecked: UnsafeRawPointer(span._start),
+      byteCount: span.count &* MemoryLayout<Element>.stride
+    )
+  }
+}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 6.1, *)
+extension RawSpan {
+
+  /// The number of bytes in the span.
+  ///
+  /// To check whether the span is empty, use its `isEmpty` property
+  /// instead of comparing `count` to zero.
+  ///
+  /// - Complexity: O(1)
+  @_alwaysEmitIntoClient
+  public var byteCount: Int { _count }
+
+  /// A Boolean value indicating whether the span is empty.
+  ///
+  /// - Complexity: O(1)
+  @_alwaysEmitIntoClient
+  public var isEmpty: Bool { byteCount == 0 }
+
+  /// The indices that are valid for subscripting the span, in ascending
+  /// order.
+  ///
+  /// - Complexity: O(1)
+  @_alwaysEmitIntoClient
+  public var byteOffsets: Range<Int> {
+    .init(_uncheckedBounds: (0, byteCount))
+  }
+}
+
+//MARK: extracting sub-spans
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 6.1, *)
+extension RawSpan {
+
+  /// Constructs a new span over the bytes within the supplied range of
+  /// positions within this span.
+  ///
+  /// The returned span's first byte is always at offset 0; unlike buffer
+  /// slices, extracted spans do not share their indices with the
+  /// span from which they are extracted.
+  ///
+  /// - Parameter bounds: A valid range of positions. Every position in
+  ///     this range must be within the bounds of this `RawSpan`.
+  ///
+  /// - Returns: A span over the bytes within `bounds`
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _extracting(_ bounds: Range<Int>) -> Self {
```

Comment:
I've now marked these with `@lifetime(self)`. The most complete annotation would probably be something like `@lifetime(copy self)`, to communicate that the lifetime is copied from the parameter's lifetime, rather than being a further borrow of the parameter. The other way to communicate it would be to explicitly consume the parameter, and thus communicate that the only possible lifetime relationship is to transfer the lifetime from the parameter to the newly-created return value.